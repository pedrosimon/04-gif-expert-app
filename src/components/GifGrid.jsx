import React, { Fragment } from "react";
import PropTypes from "prop-types";
import GifGridItem from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ category }) => {
  const { data: imgs, loading } = useFetchGifs(category);
  return (
    <Fragment>
      <h3>{category}</h3>
      {loading && <p className="animate__animated animate__flash">Cargando...</p>}
      <div className="card-grid">
        {imgs.map((img) => (
          <GifGridItem key={img.id} img={img} />
        ))}
      </div>
    </Fragment>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
