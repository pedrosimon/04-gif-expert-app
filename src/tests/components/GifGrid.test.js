import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el <GifGrid />', () => {
    const category = 'One Punch';
    
    test('El componente <GifGrid /> debe cargarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        })
        
        
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    })

    test('debería mostrar items cuando se cargan imágenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com/algo',
            title: 'un título'
        },
        {
            id: 'ABC2',
            url: 'https://google.com/algomas',
            title: 'otro título'
        }
    ]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
