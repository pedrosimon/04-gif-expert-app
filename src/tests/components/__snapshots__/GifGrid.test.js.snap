// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> El componente <GifGrid /> debe cargarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
